src/gsi/GSIReactionCatalysis.cpp

```diff
@@ -73,21 +72,40 @@ class GSIReactionCatalysis : public GSIReaction
             args.s_iter_reaction->parseError(
                 "A rate law must be provided for this reaction!");
         }
+
+         // Check for charge and mass conservation
```

Comment:
Code like this can be refactored to be cleaner and more memory efficient by having a utility function in the base class like
```c++
template <typename VecType>
double reactionDelta(const VecType& vec)
{
    double delta = 0.0;
    for (auto i: m_reactants) delta += vec[i];
    for (auto i: m_products) delta -= vec[i];
    return delta;
}
```
Then, this whole code block could become simply
```c++
for (int i = 0; i < ne; ++i)
    m_conserves &= (reactionDelta(args.s_thermo.elementMatrix().col(i)) == 0);
```
which I think is simpler and easier to understand at a glance.

Reply:
This whole class needs to be inherited from the Reaction class in kinetics. 
Considering your recommendation above, indeed the code looks better, but I am not sure I can fix this for 1.0.0, especially since I need to write tests to validate the code. Can we make it an issue for 1.1? In that version, I am planning to include detailed chemistry models, where the conservation checks for reactions need to be more advanced.

Reply:
OK, yes sounds good.